### PR TITLE
[C#] fix lambdas breaking try/catch

### LINKED ISSUE #754: [C#] lambdas break try/catch highlighting
public class Test {
        Func<int, int> a;
        void TestMe () {
            a = (progress) => progress;

            try
            {
        
            }
            catch (InvalidOperationException ex)
            {
        
            }
            finally {
                
            }
        }
    }

`catch` and `finally` are scoped as `invalid.illegal`, but they are perfectly valid here.

### PR DESCRIPTION
**Pull request by keith-hall**
fixes #754

### CODE REVIEW

File: C#/C#.sublime-syntax [hunk 1]
```diff
@@ -1541,7 +1546,7 @@ contexts:
 
   finally_block:
     - meta_scope: meta.block.trycatch.cs
-    - match: (finally)\b\s*
+    - match: finally\b
```
Comment (FichteFoll) [inline C#/C#.sublime-syntax]: Are you sure removing the `\b` anchor doesn't introduce a regression? I didn't check how the context flow goes.

Comment (keith-hall) [inline C#/C#.sublime-syntax]: I didn't remove it :) I only removed `\s*`, which meant that whitespace after `finally` was getting scoped as `keyword.control.trycatch.finally.cs`

Comment (FichteFoll) [inline C#/C#.sublime-syntax]: ... right, ignore me

**Comment by wbond** [comment]
Thanks!

**Pull request merged**